src/lib/comms/include/sol-connman.h

```diff
@@ -0,0 +1,382 @@
+ */
+int sol_connman_service_get_network_address(
+    const struct sol_connman_service *service, struct sol_network_link **link);
+
+/**
+ * @brief Gets the service strength
+ *
+ * This function gets the service strength for the connman service,
+ * after the service is to give by service monitor.
+ *
+ * @param service The service structure which the service strength is
+ * desired
+ *
+ * @return 0-100 on success, -errno on failure.
+ */
+int32_t sol_connman_service_get_strength(
+    const struct sol_connman_service *service);
+
+/**
+ * @brief Gets the global connection state of system
+ *
+ * This function gets the global connection state for system.
+ *
+ * @return The state of the global connection state on success,
+ * SOL_CONNMAN_STATE_UNKNOWN on failure
+ */
+enum sol_connman_state sol_connman_get_state(void);
+
+/**
+ * @brief Sets the global connection state to offline
+ *
+ * This function sets the global connection state to offline,
+ * enter offline Mode too.
+ *
+ * @param enabled The value is set for offline on/off
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int sol_connman_set_radios_offline(bool enabled);
+
+/**
+ * @brief Gets the global connection state of offline
+ *
+ * This function gets the global connection state of offline.
+ *
+ * @return True if the offline is enabled, false otherwise.
+ */
+bool sol_connman_get_radios_offline(void);
+
+/**
+ * @brief Connect the service
+ *
+ * This function connects the service, after the service is to give by service monitor.
+ * Since the connman function is asynchronous, the return is not connection error,
+ * but just some dispatching/immediate error. The actual state change will be notified
+ * via sol_connman_add_service_monitor() callbacks. This callback must be added BEFORE
+ * the sol_connman_service_connect() is called to ensure no messages are lost.
+ *
+ * @param service The service structure which the connection is desired
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int sol_connman_service_connect(struct sol_connman_service *service);
+
+/**
+ * @brief Disconnect the service
+ *
+ * This function disconnects the service.
+ * Since the connman function is asynchronous, the return is not disconnection error,
+ * but just some dispatching/immediate error. The actual state change will be notified
+ * via sol_connman_add_service_monitor() callbacks. This callback must be added BEFORE
+ * the sol_connman_service_disconnect() is called to ensure no messages are lost.
+ *
+ * @param service The service structure which the disconnection is desired
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int sol_connman_service_disconnect(struct sol_connman_service *service);
+
+/**
+ * @brief Adds a monitor for the updated connman services
+ *
+ * To monitor the property of services, it gives information
+ * about the services. The callback is used to provide the information.
+ * sol_connman_add_service_monitor will call sol_connman_lazy_init
```

Comment:
do not cite lazy_init, which is an internal function. Just say this will trigger all the required activations.